### PR TITLE
2.x: Fix wrong comments in Functions.java “Function3” -> “BiFunction”

### PR DESCRIPTION
**Pull request by ggikko**
I'm using this library in my company(KakaoBank in Korea) and interested in this library.. so I read whole code these days. I feel really thanks for all people who make awesome library and I just wanted to contribute this library.

### CODE REVIEW

File: src/main/java/io/reactivex/internal/functions/Functions.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
 import io.reactivex.schedulers.Timed;
 
 /**
- * Utility methods to convert the Function3..Function9 instances to Function of Object array.
+ * Utility methods to convert the BiFunction..Function9 instances to Function of Object array.
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/functions/Functions.java]: Actually, `BiFunction, Function3..Function9` would be better.

**Comment by ggikko** [comment]
I also added fail test case toFunction about Functions.java

File: src/test/java/io/reactivex/internal/functions/FunctionsTest.java [hunk 2]
```diff
@@ -169,6 +169,54 @@ public Integer apply(Integer t1, Integer t2, Integer t3, Integer t4, Integer t5,
         }).apply(new Object[20]);
     }
 
+    @Test(expected = NullPointerException.class)
+    public void biFunctionFail() throws Exception {
```
Comment (akarnokd) [inline src/test/java/io/reactivex/internal/functions/FunctionsTest.java]: Please add `@SuppressWarnings({"unchecked", "rawtypes"})` to all these new methods.

**Review by akarnokd — CHANGES_REQUESTED** [review]

File: src/main/java/io/reactivex/internal/functions/Functions.java (hunk 1 above)
Comment (ggikko) [reply src/main/java/io/reactivex/internal/functions/Functions.java]: I agree.

File: src/test/java/io/reactivex/internal/functions/FunctionsTest.java (hunk 2 above)
Comment (ggikko) [reply src/test/java/io/reactivex/internal/functions/FunctionsTest.java]: Thank you for comment.

**Review by akarnokd — APPROVED** [review]

**Pull request merged**